### PR TITLE
MM-11709 Add overview to plugins plus other minor updates

### PR DESCRIPTION
**Pull request by jwilander**

### CODE REVIEW

File: site/content/extend/plugins/overview/_index.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+### Manifest
+The plugin manifest provides required metadata about the plugin, such as name and ID. It is defined in JSON or YAML. This is `plugin.json` in both the [sample](https://github.com/mattermost/mattermost-plugin-sample/blob/master/plugin.json) and [demo](https://github.com/mattermost/mattermost-plugin-demo/blob/master/plugin.json) plugins.
+
+See the [manifest reference](/extend/plugins/manifest-reference) for more information.
+
+### Server
+The server component of a plugin is written in Go and runs as a subprocess of the Mattermost server process. The Go code is centered around extending the [MattermostPlugin](https://godoc.org/github.com/mattermost/mattermost-server/plugin#MattermostPlugin) struct that contains an [API](/extend/plugins/server/reference/#API) and allows for the implementation of [Hook](/extend/plugins/server/reference/#Hooks) methods that enable the plugin to interact with the Mattermost server.
```
Comment (lieut-data) [inline site/content/extend/plugins/overview/_index.md]: grammar nit: `centre on` vs. `centre around`, but I suspect both of these would be unfamiliar to non-native English speakers. Perhaps just simply, `The Go code extends the MattermostPlugin struct that contains ...`

Comment (jwilander) [reply site/content/extend/plugins/overview/_index.md]: Good point, will update

File: site/content/extend/plugins/overview/_index.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+The sample plugin has a [shell of an implemented PluginClass](https://github.com/mattermost/mattermost-plugin-sample/blob/master/webapp/src/index.js) while the demo plugin [contains a more complete example](https://github.com/mattermost/mattermost-plugin-demo/blob/master/webapp/src/plugin.jsx).
+
+The desktop app is a shim of the web app, meaning any plugin that works in the web app will also work in the desktop app.
+
+Read more about the web app component of plugins [here](/extend/plugins/webapp).
+
+### Mobile App
+Currently there is no mobile app component of plugins but rest assured that it is planned and coming in the future.
+
+---
```
Comment (jasonblais) [inline site/content/extend/plugins/overview/_index.md]: Wondering if there's a better way to separate the last sentence from the mobile plugins section, as currently the two look related.

image

File: site/content/extend/plugins/overview/_index.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+If you'd like to understand how plugins work under the covers, [see the contributor documentation on plugins](/contribute/server/plugins).
```
[the remark this answers is not in the document]
Comment (jasonblais) [reply site/content/extend/plugins/overview/_index.md]: (Though I'm 0/5 if `backend` is obvious for non-English native speakers)

File: site/content/extend/plugins/overview/_index.md (hunk 2 above)
Comment (jwilander) [reply site/content/extend/plugins/overview/_index.md]: There's supposed to be a divider there, not sure why it's not showing up

File: site/content/extend/plugins/overview/_index.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (jwilander) [reply site/content/extend/plugins/overview/_index.md]: Backend typically means server side which isn't accurate but I can try some other wording

File: site/content/extend/plugins/overview/_index.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+The sample plugin has a [shell of an implemented PluginClass](https://github.com/mattermost/mattermost-plugin-sample/blob/master/webapp/src/index.js) while the demo plugin [contains a more complete example](https://github.com/mattermost/mattermost-plugin-demo/blob/master/webapp/src/plugin.jsx).
+
+The desktop app is a shim of the web app, meaning any plugin that works in the web app will also work in the desktop app.
+
+Read more about the web app component of plugins [here](/extend/plugins/webapp).
+
+### Mobile App
+Currently there is no mobile app component of plugins but rest assured that it is planned and coming in the future.
```
[the remark this answers is not in the document]
Comment (jwilander) [reply site/content/extend/plugins/overview/_index.md]: Sure, will update

File: site/content/extend/plugins/overview/_index.md (hunk 2 above)
Comment (jasonblais) [reply site/content/extend/plugins/overview/_index.md]: Looks like it requires four dashes for the divider to show. There were only three.

I added one more, hope the direct commit is okay (easier than asking you to add one more dash)

**Pull request merged**